src/script/ml.ts

```diff
@@ -105,32 +102,27 @@ export async function trainModel(): Promise<boolean> {
 
   const tensorFeatures = tf.tensor(features);
   const tensorLabels = tf.tensor(labels);
-
   const nn: LayersModel = createModel();
-
-  trainingTimerPromise = new Promise(resolve => {
-    setTimeout(() => {
-      resolve(true);
-    }, 2500);
-    // Promise resolves after 2.5 sec, making training take at least 2.5 sec from users perspective
-    // See "finishedTraining" function to see how this works
-  });
-
-  const onTrainEnd = () => finishedTraining();
+  const totalNumEpochs = get(settings).numEpochs;
 
   nn.fit(tensorFeatures, tensorLabels, {
-    epochs: get(settings).numEpochs,
+    epochs: totalNumEpochs,
     batchSize: 16,
     validationSplit: 0.1,
-    callbacks: { onTrainEnd }, // onEpochEnd <-- use this to make loading animation
+    callbacks: {
+      onTrainEnd,
+      onEpochEnd: (epoch: number) => {
+        // epoch only increases up to totalNumEpochs - 1
```

Comment:
Can we change this comment to something like 'epochs indexed at 0', otherwise it looks like a bug.